perl/lib/Utils/DbSnp2FormatInfo.pm

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/perl
+use 5.10.0;
+use strict;
+use warnings;
+use DDP;
+
+# Take a DbSNP 2 VCF file, and for each row, split the INFO field's FREQ data into separate INFO fields for each population
+package Utils::DbSnp2FormatInfo;
+
+our $VERSION = '0.001';
+
+use File::Basename qw/basename/;
+
+use Mouse 2;
+use namespace::autoclean;
+use Path::Tiny qw/path/;
+
+use Seq::Tracks::Build::LocalFilesPaths;
+
+# Exports: _localFilesDir, _decodedConfig, compress, _wantedTrack, _setConfig, logPath, use_absolute_path
+extends 'Utils::Base';
+
+sub BUILD {
+  my $self = shift;
+
+  my $localFilesHandler = Seq::Tracks::Build::LocalFilesPaths->new();
+
+  my $localFilesAref = $localFilesHandler->makeAbsolutePaths(
+    $self->_decodedConfig->{files_dir},
+    $self->_wantedTrack->{name},
+    $self->_wantedTrack->{local_files}
+  );
+
+  $self->{_localFiles} = $localFilesAref;
+}
+
+# TODO: error check opening of file handles, write tests
+sub go {
+  my $self = shift;
+
+  my @output_paths;
+
+  for my $input_vcf ( @{ $self->{_localFiles} } ) {
+    if ( !-e $input_vcf ) {
+      $self->log( 'fatal', "input file path $input_vcf doesn't exist" );
+      return;
+    }
+
+    my ( $err, $isCompressed, $in_fh ) = $self->getReadFh($input_vcf);
+
+    $isCompressed ||= $self->compress;
+
+    if ($err) {
+      $self->log( 'fatal', $err );
+      return;
+    }
+
+    my $base_name = basename($input_vcf);
+    $base_name =~ s/\.[^.]+$//; # Remove last file extension (if present)
+    $base_name
+      =~ s/\.[^.]+$//; # Remove another file extension if it's something like .vcf.gz
+
+    my $output_vcf_data = $base_name . "_vcf_data.vcf" . ( $isCompressed ? ".gz" : "" );
+    my $output_vcf_header =
+      $base_name . "_vcf_header.vcf" . ( $isCompressed ? ".gz" : "" );
+    my $output_vcf = $base_name . "_processed.vcf" . ( $isCompressed ? ".gz" : "" );
+
+    $self->log( 'info', "Reading $input_vcf" );
+
+    my $output_header_path =
+      path( $self->_localFilesDir )->child($output_vcf_header)->stringify();
+    my $output_data_path =
+      path( $self->_localFilesDir )->child($output_vcf_data)->stringify();
+    my $output_path = path( $self->_localFilesDir )->child($output_vcf)->stringify();
+
+    if ( ( -e $output_data_path || -e $output_header_path || -e $output_path )
+      && !$self->overwrite )
+    {
+      $self->log( 'fatal',
+        "Temp files $output_data_path, $output_header_path, or final output path $output_path exist, and overwrite is not set"
+      );
+      return;
+    }
+
+    my $output_data_fh = $self->getWriteFh($output_data_path);
+
+    $self->log( 'info', "Writing to $output_data_path" );
+
+    # Store populations seen across the VCF
+    my %populations;
+
+    my @header_lines;
+    while (<$in_fh>) {
+      chomp;
+
+      # If it's a header line
+      if (/^#/) {
+        push @header_lines, $_;
+        next;
+      }
+
+      # If it's an INFO line
+      if (/FREQ=/) {
```

Comment:
Yeah, I found this curious too. This is actually a clever solution, unless I'm missing something. ChatGPT came up with this, and I ended up keeping it because it was cute and quite elegant.

It checks rows for the FREQ=. If it doesn't find that, there's nothing to do but add the row as is. FREQ isn't guaranteed to be in the INFO field. It is also not found any other field, and never will be.

Then we split on ";". That will result in 1 field that contains everything up until the first INFO value, and the rest of the info values. In that field that has FREQ=, we extract the FREQ=(.*) value, expand the individual values as INFO fields, join by ";" to the first field.

I would have written this by splitting each field by "\t", then search INFO, but didn't see any downsides with this approach. FREQ is guaranteed to never appear 2x, though it could potentially appear 0 times, and this handles that case :)

Reply:
INFO is a positional field so what Alex is searching for is some value within that field. I see the logic. I would use index to find "FREQ=" rather than a regular expression but that's just my preference.

My reading of the regexps is that they would catch some non-numeric characters and your tests suggest it catches trailing stuff. I'm not sure how important that is for you.